docs/source/developers/python.rst

```diff
@@ -335,6 +335,11 @@ virtualenv) enables cmake to choose the python executable which you are using.
 
 For any other C++ build challenges, see :ref:`cpp-development`.
 
+In case you may need to rebuild the C++ part due to errors in the process it is
+advisable to delete the content of the build folder with command ``rm -rf *``
+from the build folder itself. If the build has passed successfully and you need to rebuild
```

Comment:
I would advocate `git clean -Xfd .` (which removes any git-ignored file) rather than `rm -rf *`. You don't have to run it from the exact directory.

Reply:
The `rm -rf *` could also be changed to `rm -rf /cpp/build` to make this more explicit (and without needing to be run from that directory). 
I think a possible advantage of `rm` vs `git clean` is that the clean command will remove everything else as well (eg if you have multiple build directories (although a beginning contributor won't have this), or sphinx doctrees / doc output, etc)

Reply:
(of course, assuming that you would then run `git clean` from the root of the repo)

Reply:
Again, I'm assuming you're working inside the `python` directory :-)

But `git clean -Xfd .`, at least, won't provoke any disasters if you run it from the wrong directory!

Reply:
Note that it would be `git clean -Xfd ../cpp/build` then, in case you are working from `arrow/python`, since this part is about the C++ build.

Reply:
Hmm, that was not obvious to me. I would generally delete the build folder itself, rather than delete its contents.

Reply:
What if we add both possible ways? The beginning contributor would pick one if it will be written understandably or try both. For advanced users it is clear that both cases work so no harm done?

Reply:
It's generally puzzling for people if you offer two different ways of doing the same thing. Also, here, it won't be doing exactly the same thing, which can be even more confusing. I would generally advocate `git clean -Xfd` because it should clean all build leftovers, including ones that may be outside of the build directory (it shouldn't happen in C++, but it can happen in Python).

Reply:
> I would generally delete the build folder itself, rather than delete its contents.

Yes, that was also the reason I mentioned to mention `rm -rf /cpp/build` instead of `rm -rf *` from inside that directory.

I agree that we can recommend the `git clean` way instead of `python setup.py clean` (your other inline comment below). But for explicitly deleting the C++ build directory (which is the topic here), wouldn't `rm -rf /cpp/build` be more explicit (also for understanding what is actually done / needed in this case)?

Reply:
Ah, agreed. Yes, in this case the `rm -rf <build dir>` is more explicit (also, for advanced uses you may have several build directories).

Reply:
General comment: The typical workflow to reset a build when using cmake is to remove the `<build dir>` not its contents.